packages/core/src/interfaces/editor.ts

```diff
@@ -10,6 +10,7 @@ import { CoreEventHandlers } from '../events';
 
 export type Options = {
   onRender: React.ComponentType<{ render: React.ReactElement }>;
+  onBeforeMoveEnd: (targetNode: Node, currentNode: Node) => void;
```

Comment:
Based on the function call in the `actions` file above, this should be `onBeforeMoveEnd: (newParentNode: Node, targetNode: Node)`, right? 

Perhaps we should also include the existingParentNode as well? So something like: `onBeforeMoveEnd: (targetNode: Node, newParentNode: Node, existingParentNode: Node)`. This way related to your issue #318, when a user moves a Node, you can update the width of the Node that was moved, and you can also update the widths of the children of the existing and new parent node.